rfcs/header-navigation-component.md

```diff
@@ -0,0 +1,56 @@
+# Header Navigation Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {HeaderNavigation} from 'baseui/header-navigation';
+import {Button, KIND} from 'baseui/button';
+
+export default () => <HeaderNavigation mainElement={<span>Uber</span>}>
+  <a href="/a">Link One</a>
+  <a href="/b">Link Two</a>
+  <Button onClick={() => console.log('clicked')} kind={KIND.primary}>Get started</Button>
+</HeaderNavigation>;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {HeaderNavigation} from 'baseui/header-navigation';
+import {Button, KIND} from 'baseui/button';
+import {StatefulMenu as Menu} from 'baseui/menu';
+const ITEMS = [{label: 'menu item 1'}, {label: 'menu item 2'}];
+
+export default () => <HeaderNavigation mainElement={<Menu items={ITEMS} />}
+  overrides={{
+   Root: {
+     style: {
+        justifyContent: 'space-between'
+     }
+   }
+  }}>
+  <a href="/a">Link One</a>
+  <a href="/b">Link Two</a>
+</HeaderNavigation>;
+```
+
+## Exports
+
+* `HeaderNavigation`
+* `StyledRoot`
+
+## `HeaderNavigation` API
+
+* `mainElement: React$Node` - Optional
```

Comment:
Is `mainElement` too generic? If it's always going to be a logo/wordmark maybe just naming it `logo`?

Reply:
I thought about it. By default it's just a logo. However makes sense to put here some hamburger menu or other active element of header as it usually happens
@nguyensomniac what do you think?

Reply:
homeElement sounds better